In [1]:
import pandas as pd
import networkx as nx

In [2]:
df = pd.read_csv('graph/final_fr.csv')

In [3]:
df.head()

viewers       streamer                      category  counts
0        00000kim         Kamet0                 Just Chatting      16
1         0000_89  AntoineDaniel  Fall Guys: Ultimate Knockout      17
2  0000kaizer0000         Kamet0                 Just Chatting      38
3  0000kaizer0000         Kamet0             Teamfight Tactics      16
4  0000kaizer0000          Ponce                Special Events      13

In [4]:
df.describe()

counts
count  2.117840e+06
mean   2.442747e+01
std    2.746864e+01
min    1.100000e+01
25%    1.200000e+01
50%    1.500000e+01
75%    2.500000e+01
max    6.920000e+02

In [5]:
# filter out .99999 quantile
# df = df[df['weight'] < df['weight'].quantile(.9999)]
# df = df[df['weight'] > 15]

In [6]:
df.shape

(2117840, 4)

In [11]:
# create a directed graph
G = nx.from_pandas_edgelist(df[:100000], 'viewers', 'streamer', edge_attr="counts", create_using=nx.DiGraph())

In [8]:
from networkx.algorithms.community import greedy_modularity_communities

In [1]:
communities = greedy_modularity_communities(G, weight='counts', resolution=0.5, n_communities=20)

2 minutes et 37 secondes pour 50000 données (2.5% du dataset)

au delà de 30 minutes pour 100000 données (5% du dataset)